lib/sentry/transport.ex

```diff
@@ -73,12 +76,12 @@ defmodule Sentry.Transport do
 
         {:retry_after, delay_ms}
 
-      {:ok, status, headers, _body} ->
+      {:ok, status, headers, body} ->
         error_header =
           :proplists.get_value("X-Sentry-Error", headers, nil) ||
             :proplists.get_value("x-sentry-error", headers, nil) || ""
 
-        {:error, "Received #{status} from Sentry server: #{error_header}"}
+        {:error, {:server_error, {status, error_header, body}}}
```

Comment:
This is the biggest question - do we want to return only the headers that match x-sentry-error or all headers @whatyouhide?

Reply:
All headers, it's pretty much free to return all of them and I can't think of a good reason to only return `x-sentry-*` headers.